---
title: "Ghana BINs draft manuscript"
affiliations:
  - id: UoO
    name: Department of Biology, University of Oxford
    city: Oxford
    country: United Kingdom
  - id: UoG
    name: University of Ghana
author: 
  - name: 
      given: David 
      family: Hemprich-Bennett
    affiliation: 
      - ref: UoO
  - name: 
      given: Abigail
      family: Bailey
    affiliation: 
      - ref: UoO
  - name: 
      given: Ezekiel
      family: Donkor
    affiliation: 
      - ref: UoG
  - name: 
      given: Bernard
      family: Evans
    affiliation: 
      - ref: UoG
  - name: 
      given: Naana
      family: Acquaah
    affiliation: 
      - ref: UoG
  - name: 
      given: Charles
      family: Godfray
    affiliation: 
      - ref: UoO
  - name: 
      given: Owen
      family: Lewis
    affiliation: 
      - ref: UoO
  - name: 
      given: Fred
      family: Aboagye-Antwi
    affiliation: 
      - ref: UoG
  - name: 
      given: Talya
      family: Hackett
    affiliation: 
      - ref: UoO
format: 
  pdf: default
  plos-pdf:
    keep-tex: true
  docx: default
editor: visual
date: "`r Sys.Date()`"
date-format: medium
execute:
  echo: false
  warning: false
bibliography: references.bib
---

```{r basic summary}
library(here)
library(tidyverse)
library(countrycode)

bold_and_earthcape_combined <- read_csv(
  file = here('data', 'processed_data', 
              'bold_and_earthcape_combined.csv')
) %>% janitor::clean_names() %>%
  
  mutate(
    # make the date/time columns formatted as date/time, not character
    date_time_start = dmy_hm(date_time_start),
    # make a column of just the start DATES of sampling (remove the time data)
    start_date = date(date_time_start),
    sampling_month = month(start_date, label = T),
    sampling_year = year(start_date),
    year_and_month = paste(sampling_month, sampling_year))


# get a vector of all months sampled, arranged from first to last
sampling_months <- bold_and_earthcape_combined %>%
  arrange(start_date) %>%
  filter(!is.na(start_date)) %>%
  pull(year_and_month) %>%
  unique()


# read in dataset of just the data from our BOLD projects
our_bold_data <- read_csv(
  here('data', 'processed_data', 'our_organised_bold_data.csv'))

total_nsamples <- nrow(our_bold_data)

overall_chaorichness <- read_csv(here('results', 'overall_chaorichness.csv'))

# read in data for biogeographic summary
our_big_df <- read_csv(here('data', 'processed_data', 'bold_public_bin_matches.csv'))%>%
  mutate(geographic_region = countrycode(sourcevar = country, 
                                         origin = "country.name",
                                         destination = "region")
  ) %>%
  filter(!is.na(geographic_region), 
         !is.na(order_name))

source(here('parameters.R'))

# read in basic BIN availability
our_data_with_availability <- read_csv(
  here('data', 'processed_data', 'bold_data_with_availability.csv')) %>%
  select(BIN, available)
```

## 

# Abstract

# Introduction

## General statements on the issues of biodiversity-discovery in the tropics

### There's lots of it

Like, loads. But the worst part is most of it is completely undescribed, basically unknown.

There have been many estimates of global arthropod richness, for example an article from the 2018 Annual Review of Entomology [@storkHowManySpecies2018] estimated 7 million species of terrestrial arthropods, with only 1 million of them formally named. Under this estimate, only 20% of insect species had been identified. The global dry biomass of this immense richness of insects is estimated at 300 million metric tons [@rosenbergGlobalBiomassNumber2023], a similar mass to the combined biomass of humanity and its livestock.

### A lot of it's under threat

The tropics are going through some things: climate change, widespread land-use change, introduced species, pollution, nitrification. The negative impacts of these threats can act synergistically, but with uneven impacts even for closely-related taxa [@williamsonLocalScaleTemperature2022]. The impacts of these changes are better-understood for more heavily-studied taxa such as vertebrates [@wagnerInsectDeclineAnthropocene2021]. Low levels of baseline data and high per-taxa effort of identifying any observed taxa make estimates of insect declines very difficult. As a result, the few studies on insect declines tend to be clustered in relatively species-poor, wealthy western regions such as Europe and North America [@bluthgenInsectDeclinesAnthropocene2023]

### Formally describing any of it takes fucking ages

While collection of arthropod samples is laborious, the huge number of specimens that can be collected has long been an issue [@gotelliTaxonomicWishList2004], as a bottleneck is then hit with processing and identifying each individual arthropod .

It's worthwhile, but doing so is a long and meticulous job, would be nice if we could lend a helping-hand somehow.

## DNA barcoding can be helpful for this

### Brief description of DNA barcoding

If you make a database of DNA sequences from things you've identified, you can compare unknown sequences to that database and infer what they probably came from.

### The BIN system: what it is **and what it fucking isn't**

While DNA barcodes are invaluable when comparing unknown samples to a reference database of sequences from known-species, they can have an additional purpose: cataloguing unknown samples themselves. In addition to providing a publicly-available repository of DNA barcodes with associated metadata, the Barcode of Life Database (BOLD) [@ratnasinghamBoldBarcodeLife2007] assigns a unique identifier to sequences uploaded to it, known as a Barcode Index Number or 'BIN' [@ratnasinghamDNABasedRegistryAll2013]. BINs are clusters of similar DNA sequences, and so can act as a species-proxy when a species has yet to be formally described. While possessing a formal taxonomic description and binomial name remains the gold-standard for taxa delimitation, knowing that two samples have different (or identical) BINs can be of great value for ecological study, and sampling that generates novel BINs can highlight geographic and taxonomic areas of biodiversity that are deserving of increased attention. Uploading samples to BOLD's BIN system provides a useful resource for further study, by allowing future researchers to match their sample to a previous record, with taxonomic, geographic and temporal metadata attached.

While some researchers have used BINs as proxies for species (CITE), or even as a major strand of evidence when describing new species outright [@sharkeyMinimalistRevisionDescription2021], this remains controversial at best [@meierReAnalysisData2022] due to the inherent limitations of using a single gene and unreproducible algorithm to delineate taxonomy. We here view BINs as Operational Taxonomic Units (OTUs): a valuable tool in biology and ecology where samples are classified into clusters based on shared or diverging traits.

In this form, BINs have been used to analyse ecological phenomena in the tropics [@ashfaqInsectDiversitySaharoArabian2018; @bukowskiDNABarcodesReveal2022; @dsouzaStableBaselinesTemporal2018; @dewaardExpeditedAssessmentTerrestrial2019], while simultaneously providing what is frequently the first scientific record of previously unknown taxa.

## Ghana, and Africa more generally

West African forests are designated as a biodiversity hotspot \[IN FAIRNESS IT'S WEST AFRICAN FORESTS that are designated, dunno if our sites exactly qualify...\]: a region with a high level of biodiversity, that is threatened by human actions (CITE). This issue is compounded by the relatively low proportion of the area's species yet to be formally described; hampering ecological study, estimates of species distribution, extinction and more.

In this article we describe a high-intensity arthropod-sampling campaign in Ghana, capturing (insert number of samples here), and then generating a database of BINs from the material. We analyse the data to show rates of taxonomic discovery by trapping effort, and the geographic distribution of BINs detected in this dataset. This data is publicly available for use by the scientific community.

## Closing paragraph

*(Nice summary goes here)*

# Methods

## Field sites

Working at two villages in the Volta region of Ghana (Abutia Amegame and Mafi Agorve) we intensely collected arthropod samples during monthly visits February 2019-March 2020, and then April 2021-June 2021 (pandemic happened)(Check the full sampling dates with Talya, these are the [unique sampling months]{.underline} in our dataset but they might not all have been created equal).

## Sampling

At each site, for every visit, *x* transects were set up, with *w* malaise traps, *x* CDC traps, *y* yellow pan traps and *z* heath traps per transect. Traps were left for approximately 24 hours, apart from malaise traps, whose bottles were exchanged approximately every 6 hours within the 24-hour period.

## DNA barcoding

-   Sample sorting, shipment, timeframes etc
-   Samples were DNA barcoded at the Canadian Centre for DNA Barcoding (CCDB), using their standard protocol of photographing each specimen, before performing a non-destructive DNA extraction, PCR using the CO1 'Folmer' region [@folmerDNAPrimersAmplification1994], and sequencing on a PacBio Sequel. Through the BOLD data-management platform [@ratnasinghamBoldBarcodeLife2007], samples were assigned provisional taxonomy and a BIN [@ratnasinghamDNABasedRegistryAll2013].

## Data analyses

```{r}
our_data_download_date <- file.info(here('data', 'raw_data','GCEP','bold.xlsx'))$ctime %>% 
  ymd_hms() %>%
  format(., "%d %B %Y")

# get the dates that the bold country-queries were made
country_query_dates <- list.files(here('data', 'processed_data', 'bold_queries'),
          full.names = T) %>% 
  file.info(.) %>% 
  as.tibble() %>% 
  pull(ctime) %>%
  as_datetime() %>%
  date()

# sadly trying to format a vector of dates prettily makes min/max calculations of 
# them get confused, otherwise I would do it above.


# most-abundant BINs in our dataset's availability
retention_threshold <- 200
abundant_bin_summary <- our_data_with_availability %>%
  group_by(BIN, available) %>%
  summarise(n = n()) %>%
  arrange(desc(n)) %>%
  head(retention_threshold) %>%
  group_by(available) %>%
  summarise(n_bins = n())
```

Our data was downloaded from BOLD on `r our_data_download_date`, and BOLD was queried for information on the publicly available BINs that matched BINs in our dataset.

Using the iNEXT [@chaoRarefactionExtrapolationHill2014] R package, sampling completeness was calculated for each taxonomic order and trap type used; using the rate of BIN-accumulation to estimate the sampling effort required before all potential BINs in the landscape that would be captured by a given trap-type were obtained. Calculations only took place for combinations of taxonomic order and trap-type where at least `r nbin_threshold` BINs were detected. I.e. calculations would not have taken place for spiders and malaise traps, if only `r nbin_threshold-1` spider BINs were detected in malaise traps.

### NMDS

A series of non-metric multidimensional scaling (NMDS) analyses were ran for the dataset using the 'vegan' R package [@oksanenVeganCommunityEcology2022]. Taxonomic groups were only included in the dataset if they contained a minimum of `r nmds_inclusion_threshold` samples (i.e. a taxonomic order would only be included in the order-level NMDS if the dataset contained `r nmds_inclusion_threshold` or more samples of that order). CDC traps were excluded from this analysis due to the low number of BINs obtained from them.

To analyse the extent of potential differences between the communities included in the NMDS analyses, we also ran a Permutational Multivariate Analysis of Variance Using Distance Matrices, using bray-curtis dissimilarity, using the vegan R package, and the betadispersion of the communities in each trap was calculated.

### Geographic analyses.

We queried BOLD for the available metadata all BINs in our dataset that already had public matches from other studies.The resulting dataset was used to identify countries and continents which shared many BINs with our dataset, and the taxonomic composition of those BINs. The 20 countries with the highest numbers of public BINs matching BINs in our dataset were queried between `r min(country_query_dates) %>% format(., "%d %B %Y")` and `r max(country_query_dates) %>% format(., "%d %B %Y")`, to show areas where similar communities have been detected. This is more likely to represent geographic biases in sampling than similarities in the arthropod communities themselves.

### Closing sentence

All analyses took place using R [@rcoreteamLanguageEnvironmentStatistical2023]. All code for this manuscript is available at https://github.com/hemprichbennett/ghana_bins.

# Results

## Basic summary statistics.

We detected `r bold_and_earthcape_combined %>% filter(!is.na(bin)) %>% pull(bin) %>% unique() %>% length() %>% prettyNum(., big.mark=',')` unique BINs across the `r total_nsamples %>% prettyNum(., big.mark=',')` samples sequenced as of `r our_data_download_date` (see @tbl-per-traptype-summary). For the `r nrow(overall_chaorichness)` most abundant taxonomic orders, the average estimated sampling completeness was `r round(mean(overall_chaorichness$percent_completeness),1)`%.

```{r}
#| label: tbl-per-traptype-summary
#| tbl-cap: The number of samples and BINs sequenced for each trap type used
bold_and_earthcape_combined %>% 
  filter(!is.na(type)) %>% 
  group_by(type) %>% 
  summarise(`Number of samples` = n(), 
            `Number of BINs` = length(unique(bin))) %>%
  knitr::kable(., format.args = list(big.mark = ','))
```

Of the `r retention_threshold` most-abundant BINs in our dataset, `r abundant_bin_summary %>% filter(available == 'Already publicly available') %>% pull(n_bins)` were already publicly available, `r abundant_bin_summary %>% filter(available == 'Already sequenced, no public sequences') %>% pull(n_bins)` had already been sequenced but no other representatives of that BIN were publicly available on BOLD, and `r abundant_bin_summary %>% filter(available == 'Unique to our project') %>% pull(n_bins)` BINs were unique to our project.

## Trap types

### Trap types and novel BINs

Heath traps provided the majority of samples barcoded (@tbl-per-traptype-summary), and also provided the majority of novel BINs in our analysis @fig-bin_availability.

![Public availability of BINs detected, by trap-type](../figures/trap_bin_availability.jpeg){#fig-bin_availability}

### BIN accumulation

No trap types or taxonomic orders approached completeness (@fig-type1 and @fig-type2).

![Type 1 iNEXT plot](../figures/inext_plots/type1_inext_plot.png){#fig-type1 height="20cm"}

![Type 2 iNEXT plot](../figures/inext_plots/type2_inext_plot.png){#fig-type2 height="20cm"}

### NMDS

```{r}
order_adonis <- read_csv(here('results', 'adonis', 'order_summary.csv'))
family_adonis <- read_csv(here('results', 'adonis', 'family_summary.csv'))

order_bd <- read_csv(here('results', 'adonis', 'order_betadispersion.csv'))
family_bd <- read_csv(here('results', 'adonis', 'family_betadispersion.csv'))
```

NMDS ordinations for both taxonomic Order and Family showed some overlap between the trap types (@fig-order-nmds and @fig-family-nmds). The Permutational Multivariate Analysis of Variance test showed a significant difference between trap types at both the order level (p =`r order_adonis %>% select(p.value) %>% head(1)%>% signif(5)`) and family level (p = `r family_adonis %>% select(p.value) %>% head(1)%>% signif(5)`). However betadispersion was significant for both the order (p = `r order_bd %>% select(p.value) %>% head(1) %>% signif(5)`) and family (p = `r family_bd %>% select(p.value) %>% head(1)%>% signif(5)`) analyses. *(I'll tidy those numbers up later, just not sure how to do it yet)*

![Order-level NMDS plot](../figures/nmds/order_nmds.png){#fig-order-nmds height="20cm"}

![Family-level NMDS plot](../figures/nmds/family_nmds.png){#fig-family-nmds height="20cm"}

## Geographic analyses

There were `r our_big_df %>% select(bin_uri) %>% distinct() %>% nrow()%>% prettyNum(., big.mark=',')` distinct BINs in our project which are also found publicly on BOLD. Those BINs are found across `r our_big_df %>% select(country) %>% distinct() %>% nrow()` countries in `r our_big_df %>% select(geographic_region) %>% distinct() %>% nrow()` geographic regions, as shown in @tbl-geographic_matches:

```{r}
#| label: tbl-geographic_matches
#| tbl-cap: The number of BINs found in our dataset that have been found in each major geographic region.
read_csv(here('results', 'public_data', 'orders_by_region.csv')) %>%
  rename(`Order name` = order_name) %>%
  knitr::kable()
```

```{r}
n_top_countries <- 20
```

With that in mind, @tbl-top_countries is a breakdown of the public BINs in common with our project, from the `r n_top_countries` countries which had the most BINs in common with our project:

```{r}
#| label: tbl-top_countries
#| tbl-cap: The top 20 countries in terms of sharing BINs with our project



country_rankings <- our_big_df %>%
  select(country, bin_uri) %>%
  distinct() %>%
  group_by(country) %>%
  summarise(`Number of shared BINs` = n()) %>%
  arrange(desc(`Number of shared BINs`)) %>%
  slice(1:n_top_countries) %>%
  rename(Country = country) %>%
  mutate('Ranking' = rownames(.)) %>%
  select(Ranking, Country, `Number of shared BINs`) #
  knitr::kable(country_rankings)
```

It's encouraging to see that several African countries are in the top 20, but noteworthy that Ghana is at number `r country_rankings %>% filter(Country == 'Ghana') %>% pull(Ranking)` and there are few representatives of nearby countries. Some of that will be due to varying habitat types, but it highlights just how undersampled West Africa currently is. A lot of these common BINs are also likely to be cosmopolitan/invasive taxa, rather than species only native to Ghana and the nearby area.

# Discussion

# Conclusion

# Supporting information

# Acknowledgements

All authors are members of the Target Malaria Not-for-Profit Research Consortium (www.targetmalaria.org), which is supported by a grant from the Bill & Melinda Gates Foundation and Open Philanthropy.

# References
